src/platform/user/profile/vet360/components/base/Vet360EditModal.jsx

```diff
@@ -40,6 +40,23 @@ export default class Vet360EditModal extends React.Component {
     );
   }
 
+  componentDidUpdate(prevProps) {
+    const { transactionRequest } = this.props;
+
+    // We check if our new props contain a different transactionRequest prop
+    // as we're looking for a submission error
+    const hasNewError =
+      prevProps?.transactionRequest?.error !== transactionRequest?.error;
+
+    if (!transactionRequest?.error || !hasNewError) {
+      return;
+    }
+
+    // Scroll to new error.
+    const element = document.getElementById('profile-error');
+    if (element) element.scrollIntoView({ behavior: 'smooth' });
+  }
```

Comment:
Since `Vet360EditModalErrorMessage` is only rendered with the existence of `error` and so only mounts when a new `error` is passed, could we instead implement the scroll inside of the `componentDidMount` method of `Vet360EditModalErrorMessage`?

Reply:
Hmm, I see what you're saying. @erikphansen will you be using this error message also in Profile 2.0 (albeit modified since we will no longer have the modal)? I'm asking because `Vet360EditModalErrorMessage` is a functional component that I'd need to update to accept life cycle methods either via hooks or making it a class.

Reply:
@sanlouise I know I just said on our call that this is fine for now, now that I've processed it I think I agree with Nick and it's worth making the change to `Vet360EditModalErrorMessage`  with hooks or lifecycle methods (whichever you prefer). `Vet360EditModalErrorMessage` is also used by the Address Validation modal (https://github.com/department-of-veterans-affairs/vets-website/blob/7ef9f065e1a366010c1ebb7fc2feccb36be8b5e8/src/platform/user/profile/vet360/containers/AddressValidationModal.jsx#L245-L253) so making the error message handle scrolling to itself would be a win for the Address Validation modal as well.

Reply:
We may also want to implement the scroll behavior in a React component @kelsonic and I are working on, the [`ConcealedIcon`](https://github.com/department-of-veterans-affairs/vets-website/pull/12212#discussion_r408247824) component, which is now just a functional component but I wouldn't mind making it a class.

Reply:
Yes, this would be pivotal I would say... hahaha

Reply:
Since the `<Vet360EditModalErrorMessage  />` is effectively just rendering an `<AlertBox />` under the hood, we may want to consider adding the declarative scrollIntoView-on-mount support to AlertBox in the design system 🙂

Reply:
It looks like `<AlertBox />` already supports scrollIntoView behavior when it updates, but not when it first mounts: https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/packages/formation-react/src/components/AlertBox/AlertBox.jsx#L20

Reply:
Hm, yeah, good point. I suppose modifying the `isVisible` and `scrollOnShow` props passed to `AlertBox` could do the trick, but it might be easier to do it at `Vet360EditModalErrorMessage` since its content is dynamic

Reply:
I am currently trying to update `AlertBox` in `formation-react` - like Kelson suggested I think it's worth trying to move 

```
if (this.props.isVisible && this.props.scrollOnShow) {
      this.scrollToAlert();
    }
```

also into` ComponentDidMount ` since we should be passing `isVisible` and `scrollOnShow` statically, they should always be true in `src/platform/user/profile/vet360/components/base/Vet360EditModalErrorMessage.jsx`.

Reply:
I think the idea is that AlertBox content would be relatively static and the scroll behavior would be trigged by `isVisible` being toggled. I'm not where that's used though, if anywhere, in which case it would totally fine to update

Reply:
Thanks for your insights guys! This is a much better solution! I'm now just passing `scrollOnShow` to AlertBox - and the other update is done here https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/235

Reply:
Awesome! Usually you would have to bump the `formation-react` version [here](https://github.com/department-of-veterans-affairs/vets-website/blob/063bca21feb7d396c7357127749caf4220644c60/package.json#L240) but it looks like the carrot makes it moot in this case! 🙂

Reply:
Will still need to do a `yarn upgrade @department-of-veterans-affairs/formation-react` to update the `yarn.lock` though

Reply:
🥕

Reply:
*caret hahaha 🥕🐰